src/udev/udev-event.c

```diff
@@ -574,6 +578,12 @@ static int on_spawn_io(sd_event_source *s, int fd, uint32_t revents, void *userd
                                          fd == spawn->fd_stdout ? "out" : "err", *q);
         }
 
+        /* Re-enable the event source if we did not encounter EOF */
+        if (l > 0)
+                if (sd_event_source_set_enabled(s, SD_EVENT_ONESHOT) < 0)
+                        log_device_error(spawn->device,
+                                         "Failed to reactivate IO source of '%s'", spawn->cmd);
```

Comment:
maybe move the set_enabled() code to a goto target below, to share it between both branches that need it.

i.e. down here write:

```c
        if (l == 0)
                return 0;

reenable:
        r = sd_event_source_set_enabled(s, …);
        if (r < 0) 
                    …
        return 0;
```

and then jump to "reenable" from the earlier branch.

Reply:
I have done this in the updated PR